software-engineering-policies/CloudDevelopment/CloudDevelopmentPolicy.md

```diff
@@ -0,0 +1,329 @@
+* Cost Optimization: Optimize cloud resources usage to maximize cost-effectiveness and operational efficiency.
+* Innovation: Foster a culture of experimentation and innovation by leveraging cloud-native services and capabilities.
+
+### Benefits and Challenges
+
+A Cloud First Approach offers numerous benefits, including:
+
+* Enhanced scalability and flexibility
+* Faster time to market for software solutions
+* Improved collaboration and productivity
+* Cost savings through pay-as-you-go pricing models
+
+It also presents challenges such as:
+
+* Security and compliance concerns
+* Operational complexity during migration and integration
+* Potential vendor lock-in risks
+
+## Acceptable Use for Cloud Technologies
+
+The purpose of this section is to ensure that cloud resources are utilized in compliance with relevant laws, regulations, and organizational standards, while promoting efficiency, reliability, and data security.
+
+## Network Security
+
+### Guidance on Security with the Hub and Spoke Model. See [link](https://github.com/UKHO/how-do-i/blob/3373f41de95525cf795df0d9c1aae7ada2dcea86/docs/subscriptions.md?plain=1)
+
+## Recommendations for Container Hosting / Orchestration (Rich Shawley)
+
+### See Container Policy
+
+### Never Run As Root
+
+## Requirements and Recommendations for
+
+### Auditing
+
+### Testing - limit scope to infra testing ?? hand off to test community ?? - RS
+
+### Virus Scanning
+
+### Code Analysis
+
+### Dependency Checking for Cloud Applications
+
+### Infrastructure as Code - Terraform, etc
+
+Terraform is UKHO's primary tool of choise for IaC and should be used for in all cases. Terraform provided multi-cloud provisioning and so is capable of deploying to any cloud provider.
+
+#### Setting up a new project
+
+The Terraform deployment template found here: https://github.com/UKHO/Terraform-Deployment-Template should be used to ensure consistency when starting up any new project. Contact the DDC team if you have any questions.
+
+#### Networking
+
+The DDC team are responsible for the creating and maintaining the networking spoke Terraform for each project. This is managed in a separate GitHub repository to the project terraform and can be found here: https://github.com/UKHO/Azure-Spokes
+
+#### Modules
+
+The DDC team manages a collection of centralised Terraform modules found here: https://github.com/orgs/UKHO/teams/tfmodules/repositories
+These should be used whenever possible.
+
+#### IaC Scanning (Trivy)
+
+Trivy is UKHO's Chosen IaC Scanning tool. All Infrastructure pipelines should include a Trivy Scan stage before deployment that fails the build if misconfigurations or vulnerabilities are detected.
+A Trivy Stage is included in the previously mentioned Terraform deployment template: https://github.com/UKHO/Terraform-Deployment-Template/blob/5ca2e1c18618dad0df84ef8c08aea87a64c3b94e/azure-pipelines.yml#L37
+
+### Secrets Management
```

Comment:
Can we have a default position of 'Use Managed Identities' where possible. Secrets introduce maintence cost and having Managed Identities is something Calypso is trying to move towards more to reduce downtime due to expired secrets etc.